test/m3u8.test.js

```diff
@@ -108,6 +109,65 @@ QUnit.test('parses custom tags', function(assert) {
   );
 });
 
+QUnit.test('maps custom tags', function(assert) {
+  const manifest = '#VOD-STARTTIMESTAMP:1501533337573\n';
+  let element;
+  let calls = 0;
+
+  this.parseStream.addTagMapper({
+    expression: /^#VOD-STARTTIMESTAMP/,
+    map(line) {
+      const match = /#VOD-STARTTIMESTAMP:(\d+)/g.exec(line);
+
+      calls++;
+
+      return `#INTERMEDIATE:${match[1]}`;
+    }
+  });
+
+  this.parseStream.addTagMapper({
+    expression: /^#INTERMEDIATE/,
+    map(line) {
+      const match = /#INTERMEDIATE:(.*)/g.exec(line)[1];
+      const ISOdate = new Date(Number(match)).toISOString();
+
+      calls++;
+
+      return `#EXT-X-PROGRAM-DATE-TIME:${ISOdate}`;
+    }
+  });
+
+  this.parseStream.on('data', function(elem) {
+    element = elem;
+  });
+
+  this.lineStream.push(manifest);
+  assert.ok(element, 'element');
+  assert.strictEqual(calls, 2);
```

Comment:
shouldn't this be 1 since the regex test in `mapFn` should fail for the intermediate tag mapper?

Reply:
The mapping goes `#VOD-STARTTIMESTAMP` -> `#INTERMEDIATE` -> `#EXT-X-PROGRAM-DATE-TIME` on the same line. Each line being parsed will be processed by all registered mappers.

Reply:
🤔 not sure we should allow this sort of chained tag mapping since tag mappers could be added by different users. What do you think?

Reply:
I can have it explicitly break on the first matching mapper however I think it's a valid use case if say the BC player adds some mappers and a BC user would like to transform that data further

Reply:
hmm, that's a good point. Maybe a warning would work?